From f548bc89ccb4fbd9e3b53206929d2e348294b734 Mon Sep 17 00:00:00 2001
From: Mitchell Hentges <mhentges@mozilla.com>
Date: Mon, 7 Jun 2021 19:46:17 +0000
Subject: [PATCH] Bug 1713818: Require Python 3.6+ to run Mach r=ahal

This will both:
* Ensure that developers aren't accidentally using Mach with Python 2
  in the year 2021.
* Confirm that CI tasks aren't still using Python 2 with Mach.
* Remove support for Python 3.5, which is EOL'd.

Differential Revision: https://phabricator.services.mozilla.com/D116484
---
 build/mach_bootstrap.py             | 6 +++---
 python/mozboot/mozboot/bootstrap.py | 7 -------
 2 files changed, 3 insertions(+), 10 deletions(-)

--- a/build/mach_bootstrap.py
+++ b/build/mach_bootstrap.py
@@ -170,9 +170,9 @@ def bootstrap(topsrcdir):
     # Ensure we are running Python 2.7 or 3.5+. We put this check here so we
     # generate a user-friendly error message rather than a cryptic stack trace
     # on module import.
-    major = sys.version_info[:2][0]
-    if sys.version_info < (3, 6):
-        print("Python 3.6+ is required to run mach.")
+    major, minor = sys.version_info[:2]
+    if (major == 2 and minor < 7) or (major == 3 and minor < 5):
+        print("Python 2.7 or Python 3.5+ is required to run mach.")
         print("You are running Python", platform.python_version())
         sys.exit(1)
 
--- a/python/mozboot/mozboot/bootstrap.py
+++ b/python/mozboot/mozboot/bootstrap.py
@@ -283,6 +283,13 @@ class Bootstrapper(object):
         subprocess.check_call((sys.executable, mach_binary, "install-moz-phab"))
 
     def bootstrap(self, settings):
+        if sys.version_info[0] < 3:
+            print(
+                "This script must be run with Python 3. \n"
+                'Try "python3 bootstrap.py".'
+            )
+            sys.exit(1)
+
         if self.choice is None:
             applications = APPLICATIONS
             if isinstance(self.instance, OSXBootstrapperLight):
